components/ml/org.wso2.carbon.ml.rest.api/src/main/java/org/wso2/carbon/ml/rest/api/ModelApiV10.java

```diff
@@ -187,6 +188,48 @@ public Response predict(@Multipart("modelId") long modelId, @Multipart("dataForm
         }
     }
 
+    /**
+     * Predict using a file and return predictions as a CSV.
+     */
+    @POST
+    @Path("/predictionStreams")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response streamingPredict(@Multipart("modelId") long modelId, @Multipart("dataFormat") String dataFormat,
+                            @Multipart("file") InputStream inputStream) {
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
```

Comment:
Why is "PrivilegedCarbonContext" is preferred over "CarbonContext" in this particular line?